src/styles/main.scss

```diff
@@ -0,0 +1,78 @@
+$day-size: 100px;
+$border-color: #000;
+$background-color: #eee;
+$background-color-hover: #ffbfcb;
+$border-width: 1px;
+$gap-size: 1px;
+$padding-size: 10px;
+$font-size: 30px;
+$font-family: Arial, Helvetica, sans-serif;
+$move-up: -20px;
+$calendar-width: calc(
+  (#{$day-size} * 7) + (#{$gap-size} * 6) + (#{$padding-size})
+);
+$calendar-week: (
+  'mon': 1,
+  'tue': 2,
+  'wed': 3,
+  'thu': 4,
+  'fri': 5,
+  'sat': 6,
+  'sun': 7,
+);
+
+.calendar {
+  width: $calendar-width;
+  display: grid;
+  grid-template-columns: repeat(7, $day-size);
+  gap: $gap-size;
+  padding: $padding-size;
+  margin-left: 10px;
+
+  &__day {
+    box-sizing: border-box;
+    width: $day-size;
+    height: $day-size;
+    position: relative;
+    text-align: center;
+    border: $border-width solid $border-color;
+    background-color: $background-color;
+
+    @for $day from 1 through 31 {
+      &:nth-child(#{$day})::before {
+        content: '#{$day}';
+        font-family: $font-family;
+        font-size: $font-size;
+        position: absolute;
+        top: 50%;
+        left: 50%;
+        transform: translate(-50%, -50%);
+      }
+    }
+    &:hover {
+      cursor: pointer;
+      transform: translateY($move-up);
```

Comment:
Moving elements on hover can be disruptive to the user experience, as it causes a layout shift. Consider a different interaction effect that does not affect the document flow, such as a color change or a shadow.